client/my-sites/sidebar-unified/use-site-menu-items.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestAdminMenu } from '../../state/admin-menu/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getAdminMenu } from 'state/admin-menu/selectors';
+
+function useSiteMenuItems() {
+	const dispatch = useDispatch();
+	const selectedSiteId = useSelector( getSelectedSiteId );
+
+	const menuItems = useSelector( ( state ) => {
+		const menu = getAdminMenu( state, selectedSiteId );
+		return menu ? Object.values( menu ) : [];
```

Comment:
Can `menu ? Object.values( menu ) : []` be handled in `getAdminMenu`? Is there a reason we do it here?

Reply:
Thanks for raising this.

As things stand the result of a successful `getAdminMenu` selection is an `Object`. Therefore defaulting to an array wouldn't work well.

This may change in future. That said, an empty array coerces to a truthy value so it can be a source of errors.

For example:

```js
Boolean( [] );
// => true
```

This then requires a double check in the component akin to

```
if (menuItems?.length) {

}
```

I'm not sure on this one. Let's leave for now and revisit if the implementation in the component forces our hand.

Reply:
Thanks Dave for your clarifications!
```
if (menuItems?.length) {

}
```
is far more intuitive (at least for me) than ` Object.values( menu ) : []`
That said, this no blocking and we can iterate in the future.
Thanks!